docarray/array/mixins/evaluation.py

```diff
@@ -9,36 +9,101 @@
-        on comparing the `matches` of `documents` inside the `DocumentArray.
+        If one provides a `ground_truth` DocumentArray that is structurally identical
+        to `self`, this function compares the `matches` of `documents` inside the
+        `DocumentArray` to this `ground_truth`.
+        Alternatively, one can directly annotate the documents by adding labels in the
+        form of tags with the key specified in the `label_tag` attribute.
+        Those tags need to be added to `self` as well as to the documents in the
+        matches properties.
 
-        This method will fill the `evaluations` field of Documents inside this `DocumentArray` and will return the average of the computations
+        This method will fill the `evaluations` field of Documents inside this
+        `DocumentArray` and will return the average of the computations
 
-        :param other: The groundtruth DocumentArray` that the `DocumentArray` compares to.
         :param metric: The name of the metric, or multiple metrics to be computed
-        :param hash_fn: The function used for identifying the uniqueness of Documents. If not given, then ``Document.id`` is used.
-        :param metric_name: If provided, the results of the metrics computation will be stored in the `evaluations` field of each Document. If not provided, the name will be computed based on the metrics name.
-        :param strict: If set, then left and right sides are required to be fully aligned: on the length, and on the semantic of length. These are preventing
+        :param ground_truth: The ground_truth `DocumentArray` that the `DocumentArray`
+            compares to.
+        :param hash_fn: The function used for identifying the uniqueness of Documents.
+            If not given, then ``Document.id`` is used.
+        :param metric_name: If provided, the results of the metrics computation will be
+            stored in the `evaluations` field of each Document. If not provided, the
+            name will be computed based on the metrics name.
+        :param strict: If set, then left and right sides are required to be fully
+            aligned: on the length, and on the semantic of length. These are preventing
             you to evaluate on irrelevant matches accidentally.
+        :param label_tag: Specifies the tag which contains the labels.
         :param kwargs: Additional keyword arguments to be passed to `metric_fn`
-        :return: The average evaluation computed or a list of them if multiple metrics are required
+        :return: The average evaluation computed or a list of them if multiple metrics
+            are required
         """
+        if len(self) == 0:
+            raise ValueError('It is not possible to evaluate an empty DocumentArray')
+        if ground_truth and len(ground_truth) > 0 and ground_truth[0].matches:
+            ground_truth_type = 'matches'
+        elif label_tag in self[0].tags:
```

Comment:
what happens here if `label_tag` is None? is `label_tag` really Optional?

Reply:
ah sorry, yes it should not be optional